### PR TITLE
MM-26757 Run server tests on non-default port

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR modifies server run unit tests to launch the server on a non-default port. This avoids tests failing if a server instance is already bound to the default port.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-26889

### CODE REVIEW

File: cmd/mattermost/commands/server_test.go [hunk 1]
```diff
@@ -15,6 +15,10 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+const (
+	UnitTestListeningPort = ":8066"
```
Comment (lieut-data) [inline cmd/mattermost/commands/server_test.go:19]: Would we consider a random available port instead? I have bee known to use `:8066` to run a parallel server instance :)

Comment (wiggin77) [reply cmd/mattermost/commands/server_test.go]: Fair point.  Detecting available port would be a bit more involved and probably not worth it, so I switched to a port very unlikely to be used.  Thanks.

Comment (agnivade) [reply cmd/mattermost/commands/server_test.go]: My apologies for an unsolicited comment. But if we set port to 0, kernel will assign a random port by itself, and then we can get query the network address to get the assigned port number if we want it in future.

**Comment by wiggin77** [comment]
@agnivade did you have a comment?

**Comment by lieut-data** [comment]
I think it got buried in a resolved thread: https://github.com/mattermost/mattermost-server/pull/15075#discussion_r457606713.

**Comment by agnivade** [comment]
@wiggin77 https://github.com/mattermost/mattermost-server/pull/15075#discussion_r457606713

File: cmd/mattermost/commands/server_test.go (hunk 1 above)
Comment (wiggin77) [reply cmd/mattermost/commands/server_test.go]: Done

**Pull request merged**